workflow.md

```diff
@@ -40,50 +52,61 @@
+Pos.Communication.Protocol`). Issue identifier is not necessary only
+in few cases, specifically:
+* In merge commits.
+* If commit doesn't modify AST of the program. -- **TODO** @georgeee,
+  I am not sure this is correct, let's discuss. I remember there was a
+  list of cases, but I can't find it of course.
 
 ## Pull requests
 
-When you finish your work in your branch, then you should create a pull request
-to your project. Create a PR to `master` branch (or any other branch if you were
-explicitly asked). Please assign a reviewer (if you don't know a person then
-project manager would do it himself). Do not merge your own PRs!
+When you finish your work in your branch, you should create a pull
+request to your project. Create a PR to `master` branch or another
+appropriate branch. Please assign a reviewer (if you don't know a
+person then project manager would do it himself). You can merge your
+own PR only if it's approved and all mandatory CI checks pass.
 
 Please describe in PR what have you done, if it is tested or not, any additional
 information.
 
+You can also create a PR before your work is finished, but in this
+case please add a `wip` label to it.
+
 ## Merging Pull requests
 
-After review by another team member, the teammate merging the PR should merge
-`master` into the pull request's branch and then the PR branch into `master`,
-with the `--no-ff` option enabled.
+After review by another team member, the teammate merging the PR
+should merge `master` into the pull request's branch to resolve
+conflicts and then the PR branch into `master`, with the `--no-ff`
+option enabled. Usually it's enough to just press `Merge` button on
+github, as long as all checks pass and there are no conflicts.
 
-Generally, the pull request's branch should be merged into master , but if is
-clear that rebasing this branch onto master is better (for instance, it is a
-trivial change which doesn't deserve branching) then a rebase can be done
-instead.
+Generally, the pull request's branch should be merged into master,
+but if it is clear that rebasing this branch onto master is better
+(for instance, it is a trivial change which doesn't deserve branching)
+then a rebase can be done instead. Use your judgement.
 
 ## Versions
 
 Versions are denoted with 3 numbers: `Maj.Min.Fix`.
 
 * `Maj` is major version which changes rarely, when something very
-  significant is changed, it can possibly break API. It may be useful
-  for marketing.
-  * Example: hardfork in `cardano-sl`.
-* `Min` is minor version, changes when new features are added,
-  but they don't break API
-  and are not so important from marketing point of view.
-  * Example: add new endpoints to server.
+  significant is changed, it can possibly break all kinds of API it
+  provides. It may be useful for marketing.
+  * Example: input endorsers are added to `cardano-sl`.
+* `Min` is minor version, changes when new features are added, can
+  break API but tries not to break too much.
```

Comment:
There are different APIs: API exposed by haskell code (modules, functions), API used for communication between nodes (TCP), web API, maybe something else. And there are different changes by their significance. For example, if someone renames one function in haskell code it is much less significant change than if someone breaks API for node-node communication (imo). The former change is visible only to small group of developers while the second one may be visible to much broader group of people (e. g. developers of alternative implementations or some custom clients). So I think it would be good if such changes corresponded to different parts of version.

Maybe we should use 4 parts instead of 3 if we don't manage to describe versioning policy using only 3 parts.